backends/platform/sdl/posix/posix.cpp

```diff
@@ -82,11 +81,54 @@ bool OSystem_POSIX::hasFeature(Feature f) {
 Common::String OSystem_POSIX::getDefaultConfigFileName() {
 	Common::String configFile;
 
-	// On POSIX type systems, by default we store the config file inside
-	// to the HOME directory of the user.
-	const char *home = getenv("HOME");
-	if (home != NULL && (strlen(home) + 1 + _baseConfigName.size()) < MAXPATHLEN) {
-		configFile = Common::String::format("%s/%s", home, _baseConfigName.c_str());
+	Common::String prefix;
+#ifdef MACOSX
+	prefix = getenv("HOME");
+#elif !defined(SAMSUNGTV)
+	const char *envVar;
+	// Our old configuration file path for POSIX systems was ~/.scummvmrc.
```

Comment:
I would suggest waiting one version before adding autoupgrade so that the previous version also works with the new location when this happens. I am thinking about users who will find a regression when playing a game (that can happen!) and while waiting for 1.8.1 (or 1.9.0) release will switch back to the previous version to play that game. Of course that will not work for users witching to an older versions but hopefully would catch some (most?) of the cases of users downgrading. If we want to be ultra-conservative we can wait a bit more.